types/convert.go

```diff
@@ -580,6 +580,10 @@ func ConvertJSONToDecimal(sc *stmtctx.StatementContext, j json.BinaryJSON) (*MyD
 
 // getValidFloatPrefix gets prefix of string which can be successfully parsed as float.
 func getValidFloatPrefix(sc *stmtctx.StatementContext, s string) (valid string, err error) {
+	if sc.InDeleteStmt && s == "" {
```

Comment:
This function is used in `StrToUint`, maybe not only delete statement?

Reply:
The  condition  `statementContext.InDeleteStmt == true` will keep it only work in delete statement. 

And in delete statement, empty string can always convert to 0 without error and warning.(according to behaviour of mysql 5.x and 8.x ).

But in other statement like select, tidb is consistent with mysql(all with warning),  so I restrict it in delete statement context.